src/parquet/column/scanner.h

```diff
@@ -114,7 +114,7 @@ class TypedScanner : public Scanner {
         return false;
       }
     }
-    *def_level = descr()->is_optional() ?
+    *def_level = !descr()->is_required() ?
       def_levels_[level_offset_] : descr()->max_definition_level();
```

Comment:
In the false branch, the output should always be zero, using `max_definition_level()` obscures that fact. I think this code would be much clearer as

```c++
*def_level = descr->max_definition_level() > 0? def_levels_[level_offset_] : 0;
```

As is, it's confusing because the leaf node can actually *be* required, but implicitly null if any of its optional ancestors is null. If that makes sense? Same comment for `*rep_level` below